src/reconcile/rec_visibility.c

```diff
@@ -247,7 +250,7 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
          * disk.
          */
         if (F_ISSET(r, WT_REC_VISIBLE_ALL) && !WT_IS_METADATA(session->dhandle) ?
```

Comment:
Why not we do if (!is_hs_store && (F_ISSET(r, WT_REC_VISIBLE_ALL) && !WT_IS_METADATA(session->dhandle) ? WT_TXNID_LE(r->last_running, txnid) : !__txn_visible_id(session, txnid)))?

Reply:
Do you want to change it for the sake of readability or functionality?

Personally I feel that the `if` statement is already getting a bit complex to read and your suggested change might not make it any better.

Reply:
I mean functionally. We should bypass the check completely if it is a history store page.

Reply:
Oh in that case let me have a look again.